src/blocks/tipset.rs

```diff
@@ -343,25 +343,28 @@ impl Tipset {
     }
     /// Returns an iterator of all tipsets
     pub fn chain(self, store: impl Blockstore) -> impl Iterator<Item = Tipset> {
-        itertools::unfold(Some(self), move |tipset| {
-            tipset.take().map(|child| {
-                *tipset = Tipset::load(&store, child.parents()).ok().flatten();
-                child
-            })
-        })
+        let mut parent_tsk = self.parents().clone();
+        std::iter::once(self).chain(std::iter::from_fn(move || {
+            if let Ok(tipset) = Tipset::load_required(&store, &parent_tsk) {
+                parent_tsk = tipset.parents().clone();
+                Some(tipset)
+            } else {
+                None
+            }
+        }))
```

Comment:
Let's either keep the inner logic the same:
```suggestion
        let mut tipset = Some(self);
        std::iter::from_fn(move || {
            tipset.take().map(|child| {
                tipset = Tipset::load(&store, child.parents()).ok().flatten();
                child
            })
        })
```
Or simply it a bit more:
```suggestion
        let mut tipset = Some(self);
        std::iter::from_fn(move || {
            let child = tipset.take()?;
            tipset = Tipset::load_required(&store, child.parents()).ok();
            Some(child)
        })
```